be/src/storage/persistent_index.cpp

```diff
@@ -743,34 +766,80 @@ Status PersistentIndex::load(const PersistentIndexMetaPB& index_meta) {
     fs::CreateBlockOptions wblock_opts({l0_index_file_name});
     wblock_opts.mode = Env::MUST_EXIST;
     RETURN_IF_ERROR(block_mgr->create_block(wblock_opts, &_index_block));
-
+    RETURN_IF_ERROR(_delete_expired_index_file(start_version));
     return Status::OK();
 }
 
 Status PersistentIndex::prepare(const EditVersion& version) {
+    _dump_snapshot = false;
     _version = version;
     return Status::OK();
 }
 
 Status PersistentIndex::abort() {
+    _dump_snapshot = false;
     return Status::NotSupported("TODO");
 }
 
 Status PersistentIndex::commit(PersistentIndexMetaPB* index_meta) {
```

Comment:
If commit failed, when to delete the file.

Reply:
> If commit failed, when to delete the file.

If commit failed, there may be two conditions:
First is we don't create a new `snapshot_file`, and we don't need to delete it but just truncate.
Second is we created a new `snapshot_file` and it will be deleted in `abort()`.
Anyway, if any error occurred between `prepare()` and `on_committed`, we should call `abort()` and do some clean-up work (maybe reload mutableIndex to rebuild L0)